connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -251,7 +251,7 @@ class SparkSession private[sql] (
           .addAllPosArguments(args.map(lit(_).expr).toImmutableArraySeq.asJava)))
     val plan = proto.Plan.newBuilder().setCommand(cmd)
     // .toBuffer forces that the iterator is consumed and closed
-    val responseSeq = client.execute(plan.build()).toBuffer.toSeq
+    val responseSeq = client.execute(plan.build()).toSeq
```

Comment:
@juliuszsompolski I saw your comment, but I want to know what bad case exists when calling `.toSeq` directly here? Thanks ~

Reply:
Is it only necessary to rely on  `.toBuffer` forces that the iterator is consumed and closed in Scala 2.12 and not  `.toseq`?

Reply:
Yes, with just `.toSeq` the iterator does not get consumed, and does not get released. I saw that ReleaseExecution doesn't get called without the `.toBuffer`. I found `.toBuffer` to be the most concise way to force that it's consumed.

Reply:
`.toSeq` returns a lazy stream, which is not consumed (and ReleaseExecute from ExecutePlanResponseReattachableIterator doesn't get called) unless someone iterates through it entirely.
Before, there were some places doing `.foreach(_ => Unit)` for the same purpose. I found `.toBuffer` to be more concise, so changes these places to `.toBuffer`.

Reply:
Just Scala 2.12 or both Scala 2.12 and 2.13?

Reply:
Back then I testes Scala 2.12, I am not sure about 2.13.

Reply:
Currently the test seems to pass. But I think the commands that had to do this forced execution may not get coverage...
If it works, it would be good to add a test to that uses one of these commands that used to do `.toBuffer` to assert that they release execution even without `toBuffer`... Unfortunately, the `ReattachableExecuteSuite` that has access to server state does not have access to client `SparkSession` to test it... So it would have to be tested in connect-client-jvm tests with some pulling out of the ExecutePlanResponseReattachableIterator and asserting that it got closed.

Reply:
Thanks for your explanation. Although I think it's safe for Scala 2.13, I'm not familiar with the `ExecutePlanResponseReattachableIterator yet`. 

If there's no clear test case to cover it now, I'll close this pr first and reopen it if I think it's possible to make this optimization after I get familiar with the `ExecutePlanResponseReattachableIterator` part.